snuba/migrations/migration.py

```diff
@@ -45,13 +46,64 @@ def backwards(self, context: Context, dry_run: bool) -> None:
         raise NotImplementedError
 
 
-class MultiStepMigration(Migration, ABC):
+class GlobalMigration(Migration, ABC):
+    """
+    Consists of one or more Python functions executed once globally.
+    """
+
+    @abstractmethod
+    def forwards_global(self) -> Sequence[RunPython]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def backwards_global(self) -> Sequence[RunPython]:
+        raise NotImplementedError
+
+    def forwards(self, context: Context, dry_run: bool) -> None:
+        if dry_run:
+            for op in self.forwards_global():
+                desc = op.description() or "No description provided"
+                print(f"Non SQL operation - {desc}")
+            return
+
+        migration_id, logger, update_status = context
+        logger.info(f"Running migration: {migration_id}")
+        if not self.is_first_migration():
+            update_status(Status.IN_PROGRESS)
+
+        for op in self.forwards_global():
+            op.execute()
+
+        logger.info(f"Finished: {migration_id}")
+        update_status(Status.COMPLETED)
+
+    def backwards(self, context: Context, dry_run: bool) -> None:
+        if dry_run:
+            for op in self.backwards_global():
+                desc = op.description() or "No description provided"
+                print(f"Non SQL operation - {desc}")
```

Comment:
I would put the func name in here too.

Reply:
I'm going to skip this for now but might rethink later. In the ones we have so far, the functions names are not particularly useful (just called forwards/backwards) which is why I added this description field.